crypto/types/types.go

```diff
@@ -5,6 +5,14 @@ import (
 	"github.com/cosmos/gogoproto/proto"
 )
 
+// JSONPubKey defines a public key that are parse from JSON file.
```

Comment:
We should add maybe a comment saying PubKey -> JSONPubKey needs a in between step.
Additionally, I would host that encoding step in the SDK instead.

Basically adding `encoding.PubKeyFromProto` and `encoding.PubKeyToProto` but to SDK types only. Then in x/staking we can just use that instead of going via comet

Reply:
hmmmmm, i tried add that in but just like u said `crypto.JSONPubKey` is just `crypto.PubKey` without proto message so i find the implementation for `PubKeyFromProto` and `PubKeyToProto` is basically the same, even then x/staking doesn't even need to use it since `crypto.PubKey` is complied with `crypto.JSONPubKey` in the genesis validator. Don't know if i'm missing anywhere else

Reply:
i think we don't need a in between step since Pubkey contains JSONPubkey already, remove the convert step in x/staking seems fine

Reply:
If you use direvtly `crypto.PubKey` I don't think export will work (it cannot marshal i believe)

Reply:
yeah the e2e failed as because of it, if we want to use cmtjson then we must use `ToCmtPubKeyInterface` to convert to comet version of `ed25519` or `secp256k1` and not our sdk types since it is not registered in their json type registry, im thinking of way to use regular json marshal for `ConsensusGenesis`